packages/commonwealth/server/controllers/server_threads_methods/get_bulk_threads.ts

```diff
@@ -205,20 +205,26 @@ export async function __getBulkThreads(
         GROUP BY TT.id
     ), contest_data AS (
       -- get the contest data associated with the thread
-          SELECT
-              TT.id as thread_id,
-              json_agg(json_strip_nulls(json_build_object(
-              'contest_id', CON.contest_id,
-              'contest_address', CON.contest_address,
-              'thread_id', TT.id,
-              'content_id', CA.content_id,
-              'start_time', CON.start_time,
-              'end_time', CON.end_time
-          ))) as "associatedContests"
-          FROM "Contests" CON
-          JOIN "ContestActions" CA ON CON.contest_id = CA.contest_id AND CON.contest_address = CA.contest_address
-          JOIN top_threads TT ON TT.id = CA.thread_id
-          GROUP BY TT.id
+        SELECT
+            TT.id as thread_id,
+            json_agg(json_strip_nulls(json_build_object(
+            'contest_id', CON.contest_id,
+            'contest_name', CM.name,
+            'contest_cancelled', CM.cancelled,
+            'contest_interval', CM.interval,
+            'contest_address', CON.contest_address,
+            'score', CON.score,
+            'thread_id', TT.id,
+            'content_id', CA.content_id,
+            'start_time', CON.start_time,
+            'end_time', CON.end_time
+        ))) as "associatedContests"
+        FROM "Contests" CON
+        JOIN "ContestManagers" CM ON CM.contest_address = CON.contest_address
+        JOIN "ContestActions" CA ON CON.contest_id = CA.contest_id 
+        AND CON.contest_address = CA.contest_address AND CA.action = 'upvoted'
+        JOIN top_threads TT ON TT.id = CA.thread_id
+        GROUP BY TT.id
```

Comment:
Currently, it'll return one entry per upvote when I think you want one entry per contest.

Should replace `CA.action = 'upvoted'` with `CA.action = 'added'`.

Also this query doesn't filter the ContestAction by thread ID. Need to add `CA.thread_id = TT.id`

Then the query will essentially be "return all contests where this thread was added as content".

Reply:
Let's get a query plan for this update. Please link one using: https://explain.dalibo.com/.

Reply:
before:
https://explain.dalibo.com/plan/452bd326f9c17c96

after:
https://explain.dalibo.com/plan/d587ab8e5f1e8ffb

Reply:
> Currently, it'll return one entry per upvote when I think you want one entry per contest.

Good catch, I didn't know how the logic worked at the time and thought we were performing front end calculations based on the number of upvotes. Now I see that we just pull the place (1st/2nd/3rd) from the Contests.score.

> Also this query doesn't filter the ContestAction by thread ID. Need to add CA.thread_id = TT.id

I think it does, we have it in the join statement on top_threads (Let me know if you are referring to something else).

Reply:
@rbennettcw Basically 'added' contest actions do not increment the `ContestActions.contest_id` field:
![image](https://github.com/hicommonwealth/commonwealth/assets/14794654/0445e8ca-c0fa-4fee-b4e2-9d2b48eade65)

Can we leave it at pulling 'upvoted' ContestActions for now? Basically the functionality will be the same, but we will pull duplicate contestActions for each thread.

Reply:
Apparently we can't get the contest ID for content– it'll always be zero. Need to use upvote.